### PR TITLE
implement HTTP/3 control stream handling

### PR DESCRIPTION
**Pull request by marten-seemann**
We'll need this if we ever want to read the SETTINGS frame. This frame is used to negotiate HTTP/3 extensions.

### CODE REVIEW

File: http3/server.go [hunk 1]
```diff
@@ -254,6 +258,33 @@ func (s *Server) handleConn(sess quic.EarlySession) {
 	}
 }
 
+func (s *Server) handleUnidirectionalStreams(sess quic.EarlySession) {
+	for {
+		str, err := sess.AcceptUniStream(context.Background())
+		if err != nil {
+			return
```
Comment (lucas-clemente) [inline http3/server.go:266]: Can/should we somehow handle the error? Log at least?

Comment (lucas-clemente) [reply http3/server.go:266]: Same for the other exit points in this fn.

File: http3/client.go [hunk 2]
```diff
@@ -127,15 +129,45 @@ func (c *client) setupSession() error {
...
 	buf := &bytes.Buffer{}
-	// write the type byte
-	buf.Write([]byte{0x0})
+	utils.WriteVarInt(buf, streamTypeControlStream)
 	// send the SETTINGS frame
 	(&settingsFrame{}).Write(buf)
-	if _, err := str.Write(buf.Bytes()); err != nil {
-		return err
-	}
+	_, err = str.Write(buf.Bytes())
```
Comment (lucas-clemente) [inline http3/client.go:135]: Ignoring short writes here?

File: http3/client.go [hunk 3]
```diff
@@ -127,15 +129,45 @@ func (c *client) setupSession() error {
...
+	return err
+}
 
-	return nil
+func (c *client) handleUnidirectionalStreams() {
+	for {
+		str, err := c.session.AcceptUniStream(context.Background())
+		if err != nil {
+			return
```
Comment (lucas-clemente) [inline http3/client.go:143]: Handle / log the error?

Comment (lucas-clemente) [reply http3/client.go:143]: Same below

**Review by lucas-clemente — APPROVED** [review]

File: http3/client.go (hunk 2 above)
Comment (marten-seemann) [reply http3/client.go:135]: Not ignoring it. But there's no temporary `Write` errors.

**Pull request merged**